[operation] OperationServiceImpl_timeoutTest.testOperationTimeoutForLongRunningRemoteOperation

```
SleepingOperation invocation failed to complete due to operation-heartbeat-timeout. Total elapsed time: 4146 ms. Last heartbeat: Tue Apr 12 05:34:56 EEST 2016. Invocation{op=com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl_timeoutTest$SleepingOperation{serviceName='null', identityHash=1369455642, partitionId=-1, replicaIndex=0, callId=0, invocationTime=1460428493339 (Tue Apr 12 05:34:53 EEST 2016), waitTimeout=-1, callTimeout=1000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=1000, firstInvocationTimeMs=1460428493339, firstInvocationTime='Tue Apr 12 05:34:53 EEST 2016', lastHeartbeatMillis=1460428496465, lastHeartbeatTime='Tue Apr 12 05:34:56 EEST 2016', target=Address[127.0.0.1]:7668, pendingResponse={VOID}, backupsExpected=0, backupsCompleted=0, connection=MockConnection{localEndpoint=Address[127.0.0.1]:7667, remoteEndpoint=Address[127.0.0.1]:7668}}
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK8/com.hazelcast$hazelcast/773/testReport/junit/com.hazelcast.spi.impl.operationservice.impl/OperationServiceImpl_timeoutTest/testOperationTimeoutForLongRunningRemoteOperation/

Failed again in maintenance pr builder

https://hazelcast-l337.ci.cloudbees.com/job/Hazelcast-pr-builder/15877/testReport/junit/com.hazelcast.spi.impl.operationservice.impl/OperationServiceImpl_timeoutTest/testOperationTimeoutForLongRunningRemoteOperation/

```
Error Message
SleepingOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2018-06-04 10:04:47.077. Start time: 2018-06-04 10:04:29.076. Total elapsed time: 18001 ms. Last operation heartbeat: 2018-06-04 10:04:32.087. Last operation heartbeat from member: 2018-06-04 10:04:45.583. Invocation{op=com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl_timeoutTest$SleepingOperation{serviceName='null', identityHash=402755160, partitionId=-1, replicaIndex=0, callId=-4, invocationTime=1528106669076 (2018-06-04 10:04:29.076), waitTimeout=-1, callTimeout=6000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=6000, firstInvocationTimeMs=1528106669076, firstInvocationTime='2018-06-04 10:04:29.076', lastHeartbeatMillis=1528106672087, lastHeartbeatTime='2018-06-04 10:04:32.087', target=[127.0.0.1]:5010, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=MockConnection{localEndpoint=[127.0.0.1]:5009, remoteEndpoint=[127.0.0.1]:5010}}
Stacktrace
java.util.concurrent.ExecutionException: SleepingOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2018-06-04 10:04:47.077. Start time: 2018-06-04 10:04:29.076. Total elapsed time: 18001 ms. Last operation heartbeat: 2018-06-04 10:04:32.087. Last operation heartbeat from member: 2018-06-04 10:04:45.583. Invocation{op=com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl_timeoutTest$SleepingOperation{serviceName='null', identityHash=402755160, partitionId=-1, replicaIndex=0, callId=-4, invocationTime=1528106669076 (2018-06-04 10:04:29.076), waitTimeout=-1, callTimeout=6000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=6000, firstInvocationTimeMs=1528106669076, firstInvocationTime='2018-06-04 10:04:29.076', lastHeartbeatMillis=1528106672087, lastHeartbeatTime='2018-06-04 10:04:32.087', target=[127.0.0.1]:5010, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=MockConnection{localEndpoint=[127.0.0.1]:5009, remoteEndpoint=[127.0.0.1]:5010}}
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.newOperationTimeoutException(InvocationFuture.java:164)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:106)
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:154)
	at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl_timeoutTest.testOperationTimeoutForLongRunningRemoteOperation(OperationServiceImpl_timeoutTest.java:208)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.hazelcast.core.OperationTimeoutException: SleepingOperation invocation failed to complete due to operation-heartbeat-timeout. Current time: 2018-06-04 10:04:47.077. Start time: 2018-06-04 10:04:29.076. Total elapsed time: 18001 ms. Last operation heartbeat: 2018-06-04 10:04:32.087. Last operation heartbeat from member: 2018-06-04 10:04:45.583. Invocation{op=com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl_timeoutTest$SleepingOperation{serviceName='null', identityHash=402755160, partitionId=-1, replicaIndex=0, callId=-4, invocationTime=1528106669076 (2018-06-04 10:04:29.076), waitTimeout=-1, callTimeout=6000}, tryCount=250, tryPauseMillis=500, invokeCount=1, callTimeoutMillis=6000, firstInvocationTimeMs=1528106669076, firstInvocationTime='2018-06-04 10:04:29.076', lastHeartbeatMillis=1528106672087, lastHeartbeatTime='2018-06-04 10:04:32.087', target=[127.0.0.1]:5010, pendingResponse={VOID}, backupsAcksExpected=0, backupsAcksReceived=0, connection=MockConnection{localEndpoint=[127.0.0.1]:5009, remoteEndpoint=[127.0.0.1]:5010}}
	... 17 more
```

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl_timeoutSlowTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl_timeoutSlowTest.java
@@ -1,0 +1,61 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.operationservice.impl;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.nio.Address;
+import com.hazelcast.spi.OperationService;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.SlowTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.spi.properties.GroupProperty.OPERATION_CALL_TIMEOUT_MILLIS;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({SlowTest.class, ParallelTest.class})
+public class OperationServiceImpl_timeoutSlowTest extends HazelcastTestSupport {
+
+    @Test
+    public void testOperationTimeoutForLongRunningRemoteOperation() throws Exception {
+        // this timeout makes the test slow. however we cannot shorten it too much
+        // as otherwise it becomes to sensitive on environmental hiccups
+        int callTimeoutMillis = 15000;
+        Config config = new Config().setProperty(OPERATION_CALL_TIMEOUT_MILLIS.getName(), "" + callTimeoutMillis);
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance hz1 = factory.newHazelcastInstance(config);
+        HazelcastInstance hz2 = factory.newHazelcastInstance(config);
+
+        // invoke on the "remote" member
+        Address remoteAddress = getNode(hz2).getThisAddress();
+        OperationService operationService = getNode(hz1).getNodeEngine().getOperationService();
+        ICompletableFuture<Boolean> future = operationService
+                .invokeOnTarget(null, new OperationServiceImpl_timeoutTest.SleepingOperation(callTimeoutMillis * 5), remoteAddress);
+
+        // wait more than operation timeout
+        sleepAtLeastMillis(callTimeoutMillis * 3);
+        assertTrue(future.get());
+    }
+}

--- a/hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl_timeoutTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl_timeoutTest.java
@@ -186,26 +186,6 @@ public class OperationServiceImpl_timeoutTest extends HazelcastTestSupport {
     }
 
     @Test
-    public void testOperationTimeoutForLongRunningRemoteOperation() throws Exception {
-        int callTimeoutMillis = 6000;
-        Config config = new Config().setProperty(OPERATION_CALL_TIMEOUT_MILLIS.getName(), "" + callTimeoutMillis);
-
-        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
-        HazelcastInstance hz1 = factory.newHazelcastInstance(config);
-        HazelcastInstance hz2 = factory.newHazelcastInstance(config);
-
-        // invoke on the "remote" member
-        Address remoteAddress = getNode(hz2).getThisAddress();
-        OperationService operationService = getNode(hz1).getNodeEngine().getOperationService();
-        ICompletableFuture<Boolean> future = operationService
-                .invokeOnTarget(null, new SleepingOperation(callTimeoutMillis * 5), remoteAddress);
-
-        // wait more than operation timeout
-        sleepAtLeastMillis(callTimeoutMillis * 3);
-        assertTrue(future.get());
-    }
-
-    @Test
     public void testOperationTimeoutForLongRunningLocalOperation() throws Exception {
         int callTimeoutMillis = 500;
         Config config = new Config();
@@ -225,7 +205,7 @@ public class OperationServiceImpl_timeoutTest extends HazelcastTestSupport {
         assertTrue(future.get());
     }
 
-    private static class SleepingOperation extends Operation {
+    public static class SleepingOperation extends Operation {
         private long sleepTime;
 
         public SleepingOperation() {

